pandas/core/indexing.py

```diff
@@ -2079,6 +2079,10 @@ def _validate_key(self, key, axis):
             arr = np.array(key)
             len_axis = len(self.obj._get_axis(axis))
 
+            if not np.issubdtype(arr.dtype, np.number):
```

Comment:
use is_integer_dtype

Reply:
This will cause a new exception when an array of float values is passed to `.iloc`.  Currently float values are allowed and the array is cast to integer dtype and will cause some regression.

Reply:
then you need to thread this in the appropriate place

Reply:
alt you could check is_numeric_dtype

Reply:
Alright I tried to reorganize it a bit, let me know if I misunderstood what you meant.